pandas/core/generic.py

```diff
@@ -5783,7 +5784,7 @@ def pct_change(self, periods=1, fill_method='pad', limit=None, freq=None,
                                   **kwargs)) - 1)
         if freq is None:
             mask = isnull(_values_from_object(self))
-            np.putmask(rs.values, mask, np.nan)
```

Comment:
``np.putmask`` *assumes* this is a view. We don't use ``inplace`` *anywhere* internally in the code, it is completely non-idiomatic.

``rs[mask] = np.nan`` looks fine to me, what is the problem?